### ISSUE TITLE
rank for PlayGamesPlatform.Instance.LoadScores + LeaderboardTimeSpan.Weekly returns all-time

### ISSUE THREAD

**Issue by frazerbw** [playgameservices/play-games-plugin-for-unity#3090]
This seems broken. I'm using the following code. 

I want to get the players rank within the weekly leaderboard, but everything I try only gives me the all-time rank. Any ideas?

`PlayGamesPlatform.Instance.LoadScores(
            WORLDWIDE_EXP_RANK_WEEKLY,
            LeaderboardStart.TopScores,
            1,
            LeaderboardCollection.Public,
            LeaderboardTimeSpan.Weekly,
            (data) =>
            {
                expWeeklyRank = data.PlayerScore.rank;
            });`

**Comment by frazerbw** [comment]
The only idea I have is I need to loop down through every single player in the Scores and count until I find the player I am looking for. But if I had +1 mil users on the highscores, that would be very very inefficient.

**Comment by frolky** [comment]
Solution: https://github.com/playgameservices/play-games-plugin-for-unity/issues/2829#issuecomment-573395075

**Comment by frazerbw** [comment]
@frolky thanks, so your solution is to change the playgameservices code? Did that work, it sounds like you ended up with issues call ReportScore after making that change?

**Comment by frolky** [comment]
@frazerbw yes, you need to make changes to the plugin code. Everything works well, the problem with ReportScore was different.

**Comment by frazerbw** [comment]
@frolky you're a legend! Thanks, will try this and report back if it works :)

**Comment by frazerbw** [comment]
@frolky this worked in my test build! People who find this thread in the future can assume it fully worked unless I am back again to say otherwise :)

**Comment by vaibhavgpt-google** [comment]
Closing this issue as part of a cleanup of older tickets. Google Play Games v1 is now deprecated. Please upgrade to version v2.1.0 or newer: https://github.com/playgameservices/play-games-plugin-for-unity/releases/tag/v2.1.0. If this issue persists on the new v2 architecture, please open a new ticket.
